sources/packages/api/src/services/student/student.service.ts

```diff
@@ -76,4 +78,44 @@ export class StudentService extends RecordDataModelService<Student> {
 
     return this.save(student);
   }
+
+  async synchronizeUserInfo(userInfo: UserInfo): Promise<any> {
+    const studentToSync = await this.getStudentByUserName(userInfo.userName);
+    //Error out if user is not found
+    if (!studentToSync) {
+      throw new Error(
+        "Not able to find a student using the username (bcsc name)",
+      );
+    }
+
+    let mustSave = false;
+
+    if (
+      userInfo.email !== studentToSync.user.email ||
+      userInfo.lastName !== studentToSync.user.lastName ||
+      userInfo.givenNames !== studentToSync.user.firstName
+    ) {
+      studentToSync.user.email = userInfo.email;
+      studentToSync.user.lastName = userInfo.lastName;
+      studentToSync.user.firstName = userInfo.givenNames;
+      mustSave = true;
+    }
+
+    let birthDate = new Date(userInfo.birthdate);
+    if (
+      birthDate !== studentToSync.birthdate ||
```

Comment:
This is going to be string comparison and not comparison of actual dates. Intentional?

Reply:
Both objects are Date types. It is supposed to be a Date/Date comparison.
Any particular reason to you believe that it would be a string comparison? Am I missing something maybe?